### PR TITLE
Supported/Case Management Models

### PR DESCRIPTION
**Pull request by AndrewCowle**
## Changes in this PR

- Added models for `Support::Enquiry` , `Support::Case` , `Support::Category` , `Support::Documents`
- Added concern for including polymorphic association of `Documents` to both `Enquiry` and `Case

## Next steps

- Merge with View and Controller work - already setup to merge into `feature/61-build-cases-views-controllers`
- Factories to support feature tests when merging

### CODE REVIEW

**Comment by AndrewCowle** [comment]
Originally setup to merge into `develop` and I had updated it with the changes from the base branch hence changes to gemfile etc included.

**Review by peterdavidhamilton — COMMENTED** [review]
Was one of the AC to test against factories? Can we add any documentation?

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ The format is based on [Keep a Changelog 1.0.0].
 
 ## [Unreleased]
 
+- Added initial models for supported case management functions.
```
Comment (peterdavidhamilton) [inline CHANGELOG.md:9]: Can you add a heading to keep the **Supported Application** work clearly separate.

File: spec/factories/support/cases.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 FactoryBot.define do
-  factory :support_case, class: 'Support::Case' do
+  factory :support_case, class: "Support::Case" do
     ref { "MyString" }
```
Comment (peterdavidhamilton) [inline spec/factories/support/cases.rb:3]: What are some example fields?

File: app/models/support.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 module Support
   def self.table_name_prefix
```
Comment (peterdavidhamilton) [inline app/models/support.rb:2]: Can you document this class method please along with the module.

**Comment by peterdavidhamilton** [comment]
Not seeing any movement on this branch, is this what @ClaudioCarmeli has merged?

**Comment by peterdavidhamilton** [comment]
@ClaudioCarmeli whilst @AndrewCowle is still to push more changes do you have any comments?

**Review by ClaudioCarmeli — APPROVED** [review]
💯

**Review by peterdavidhamilton — APPROVED** [review]

**Review by EdwinKruglov — APPROVED** [review]

**Pull request merged**